streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractSegments.java

```diff
@@ -35,18 +35,19 @@
 import java.util.TreeMap;
 
 /**
- * Manages the {@link Segment}s that are used by the {@link RocksDBSegmentedBytesStore}
+ * Manages the {@link KeyValueSegment}s or {@link TimestampedSegment} that are used by the {@link RocksDBSegmentedBytesStore}
```

Comment:
It's an antipattern for a superclass to specify what subclasses it, specifically because it's likely that the *next* time someone adds a new subclass, it's unlikely they'll remember to update this comment, so then this comment will become misleading.

Plus, the HTML doc generator will automatically create a list of known subclasses, and any IDE also makes it trivially easy to discover what implements it.

Bonus: removing these references frees you from the obligation to add the dead-code classes that they point to.

IMHO, all the javadoc really needs to say is that this class manages `{@link Segment}`s.

Reply:
Ack. Will remove the two dead code classes and remove this comment.